core-backend/sandbox/src/funcs.rs

```diff
@@ -396,8 +396,35 @@ impl<E: Ext + 'static> FuncsHandler<E> {
             .map_err(|_| {
                 ctx.trap = Some("Trapping: unable to send reply message");
                 HostError
-            });
-        result
+            })
+    }
+
+    pub fn reply_wgas(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
+        let mut args = args.iter();
+
+        let payload_ptr = pop_i32(&mut args)?;
+        let payload_len = pop_i32(&mut args)?;
+        let value_ptr = pop_i32(&mut args)?;
+        let gas_limit_ptr = pop_i32(&mut args)?;
+
+        ctx.ext
+            .with(|ext| {
+                let payload = funcs::get_vec(&ctx.memory, payload_ptr, payload_len)?;
+                let value = funcs::get_u128(&ctx.memory, value_ptr)?;
+                let gas_limit = funcs::get_u128(&ctx.memory, gas_limit_ptr)?;
```

Comment:
```suggestion
                let gas_limit = funcs::get_u128(&ctx.memory, gas_limit_ptr)?;
                let value = funcs::get_u128(&ctx.memory, value_ptr)?;
```

Gas limit aren't u128 type.
It could be thrown as it is: https://github.com/gear-tech/gear/blob/2f2fc42c21e21237c4ea6ead7a59a85d2da903d5/core-backend/sandbox/src/funcs.rs#L114

Reply:
ok, sounds good, I'll fix that
But I am wondering- why don't use ```value``` "as it is" too? why do we need a ptr for it?
as I understand we use prt for ```payload``` to cut redundant copy semantic becouse ```payload``` could be big and extra copy will hurt, but that's not not true for ```value```. It has +- same size as ptr

Reply:
We are restricted to use the WebAssembly compatible types only. There is no neither 128-bit numeric value, nor vector type (see e.g. https://docs.rs/wasmi/latest/wasmi/enum.ValueType.html).